src/targets/gpu/gemm_impl.cpp

```diff
@@ -99,180 +115,477 @@ static bool is_transposed(const shape& s)
+                    output_type,
+                    ldd,
+                    compute_type);
+    }
+#ifdef ROCBLAS_BETA_FEATURES_API
+    /**
+     * Find best rocBLAS solution:  Get list of solutions and try them all, returning the index
+     * of the fastest one.
+     */
+    int tune(context& ctx, const std::vector<shape>& input_shapes) const
+    {
+        std::vector<argument> input_args;
+        std::transform(input_shapes.begin(),
+                       input_shapes.end(),
+                       std::back_inserter(input_args),
+                       [](const shape& x) { return to_gpu(generate_argument(x)); });
+
+        // Get the solutions list in 2 rocBLAS steps:
+        // 1.  Find out how many solutions there are and allocate the array
+        // 2.  Get the solutions
+        //
+        rocblas_int list_size = 0;
+        std::vector<rocblas_int> solution_indices;
+        if(strided_batched)
+        {
+            auto common_args = create_strided_batched_args_common(ctx, input_args);
+            auto ded_args = pack(rocblas_gemm_algo_solution_index, int8_flag, nullptr, &list_size);
+
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex_get_solutions,
+                           pack_join(common_args, ded_args));
+            solution_indices.resize(list_size);
+
+            auto common_sol_args = create_strided_batched_args_common(ctx, input_args);
+            auto ded_sol_args    = pack(
+                rocblas_gemm_algo_solution_index, int8_flag, solution_indices.data(), &list_size);
+
+            rocblas_invoke(&rocblas_gemm_strided_batched_ex_get_solutions,
+                           pack_join(common_sol_args, ded_sol_args));
+        }
+        else
+        {
+            auto common_args = create_gemm_ex_args_common(ctx, input_args);
+            auto ded_args = pack(rocblas_gemm_algo_solution_index, int8_flag, nullptr, &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions, pack_join(common_args, ded_args));
+            solution_indices.resize(list_size);
+
+            auto common_sol_args = create_gemm_ex_args_common(ctx, input_args);
+            auto ded_sol_args    = pack(
+                rocblas_gemm_algo_solution_index, int8_flag, solution_indices.data(), &list_size);
+            rocblas_invoke(&rocblas_gemm_ex_get_solutions,
+                           pack_join(common_sol_args, ded_sol_args));
+        }
+
+        double bestTime   = std::numeric_limits<double>::max();
+        double first_time = -1;
+        // Initialize to default solution index
+        rocblas_int bestSol = 0;
+        for(auto sol : solution_indices)
+        {
+            // Warmup: the first few calls to an op. may not be representative since there is
+            // more time taken initializing caches, etc. so we won't time them.
+            for(rocblas_int cc = 0; cc < cold_calls; ++cc)
+            {
+                run(ctx, input_args, sol);
```

Comment:
Just call this once, dont use a loop for cold calls. Also use `run_func` not `run`.

Reply:
Changing the parameter to 1 but leaving the loop in place, for now.